### PR TITLE
fix(curriculum): regex compatibility issue for Safari

### LINKED ISSUE #41724: Did we drop support for Safari?
Looking at PR [41501](https://github.com/freeCodeCamp/freeCodeCamp/pull/41501), it seems so.

Just as a heads-up we are starting to see forum threads now from Safari users and that specific [challenge](https://www.freecodecamp.org/learn/responsive-web-design/basic-css/change-a-variable-for-a-specific-area). 

Examples:

https://forum.freecodecamp.org/t/issue-with-detection-for-penguin-belly/453902

https://forum.freecodecamp.org/t/i-cant-pass-the-test-for-some-reason-even-though-it-seems-to-me-i-havent-done-any-mistakes/453964

https://forum.freecodecamp.org/t/penguin-belly-challenge-cant-be-passed/453941

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41724

### CODE REVIEW

**Comment by raisedadead** [comment]
@freeCodeCamp/moderators can someone with safari browser pull this down and test it.

File: curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md [hunk 1]
```diff
@@ -23,10 +23,19 @@ The `penguin` class should reassign the `--penguin-belly` variable to `white`.
 
 ```js
 assert(
-  code.match(/\.penguin\s*?{[\s\S]*(?<!background-color: )--penguin-belly\s*?:\s?white\s*?;[\s\S]*}/gi)
+  code.match(/.penguin\s*?{[\s\S]*--penguin-belly\s*?:\s*?white\s*?;[\s\S]*}/gi)
 );
 ```
 
+The `peguin` class should not contain the `background-color` property
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:30]:
```suggestion
The `penguin` class should not contain the `background-color` property
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Just the one typo.  Aside from that, this LGTM.  It should be fine in Safari, because lookahead is universally supported, according to [caniuse](https://caniuse.com/?search=lookahead).

That said, I agree that we should get a confirmation from a Safari user.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1:

**Comment by ahmadabdolsaheb** [comment]
@moT01, could you double check this. I am getting errors with `Loadable` when visiting a page with editor on Safari locally. (chrome is fine)

**Comment by ojeytonwilliams** [comment]
@ahmadabdolsaheb it's possibly a silly question, but do you see the `Loadable` error in GitPod?

File: curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md [hunk 2]
```diff
@@ -23,10 +23,19 @@ The `penguin` class should reassign the `--penguin-belly` variable to `white`.
 
 ```js
 assert(
-  code.match(/\.penguin\s*?{[\s\S]*(?<!background-color: )--penguin-belly\s*?:\s?white\s*?;[\s\S]*}/gi)
+  code.match(/.penguin\s*?{[\s\S]*--penguin-belly\s*?:\s*?white\s*?;[\s\S]*}/gi)
```
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]:
```suggestion
  code.match(/\.penguin\s*?{[\s\S]*--penguin-belly\s*?:\s*?white\s*?;[\s\S]*}/gi)
```

**Review by moT01 — COMMENTED** [review]
Yea, Safari isn't working great for me either. Using `npm run develop` I get this on any page with the editor:
Screen Shot 2021-04-08 at 10 34 03 AM

And it's stuck there. If I `npm run build` it works.

I left one suggestion, the period isn't escaped there. There are few other cases not handled.


It passes if the property is outside the selector:
Screen Shot 2021-04-08 at 10 45 25 AM

It passes if extra characters are added in front of the property:
Screen Shot 2021-04-08 at 10 46 01 AM

I don't think these are terrible and necessarily need to be fixed. But if we wanted to handle these, we could possibly check the regex in conjunction with checking the color rendered? Or I have used [something like this](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/01-responsive-web-design/css-variables-skyline/part-058.md#--hints--) before to isolate the properties within a class.

I think I would add my suggestion quick, and I would be fine merging this after that if we want to get it in.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

Unless it's determined that my other concerns should be fixed. I think it's fine for now.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**